design-documents/graph-ql/coverage/catalog/configurable-options-selection.graphqls

```diff
@@ -1,21 +1,29 @@
 type ConfigurableProduct {
-    configurable_options_selection_metadata(selectedConfigurableOptionValues: [ID!]): ConfigurableOptionsSelectionMetadata @doc(description: "Metadata for the specified configurable options selection")
+    configurable_product_options_selection(configurableOptionValues: [ID!]): ConfigurableProductOptionsSelection @doc(description: "Specified configurable product options selection")
+    variants: [ConfigurableVariant] @deprecated(reason: "Use configurable_options_selection_metadata instead.") @doc(description: "An array of variants of products") @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ConfigurableVariant")
+    configurable_options: [ConfigurableProductOptions] @deprecated(reason: "Use configurable_options_selection_metadata instead.") @doc(description: "An array of linked simple product items") @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\Options")
 }
 
-type ConfigurableOptionsSelectionMetadata @doc(description: "Metadata corresponding to the configurable options selection.")
+type ConfigurableProductOptionsSelection @doc(description: "Metadata corresponding to the configurable options selection.")
 {
-    options_available_for_selection: [ConfigurableOptionAvailableForSelection!] @doc(description: "Configurable options available for further selection based on current selection.")
+    configurable_options: [ConfigurableProductOption!] @doc(description: "Configurable options available for further selection based on current selection.")
     media_gallery: [MediaGalleryInterface!] @doc(description: "Product images and videos corresponding to the specified configurable options selection.")
     variant: SimpleProduct @doc(description: "Variant represented by the specified configurable options selection. It is expected to be null, until selections are made for each configurable option.")
 }
 
-type ConfigurableOptionAvailableForSelection @doc(description: "Configurable option available for further selection based on current selection.") {
-    option_value_uids: [ID!]! @doc(description: "Configurable option values available for further selection.")
-    attribute_code: String! @doc(description: "Attribute code that uniquely identifies configurable option.")
+type ConfigurableProductOption {
+    uid: ID!
+    attribute_code: String!
+    attribute_uid: ID!
```

Comment:
Why do we need attribute_uid here? What is a use case for this? 
CC: @akaplya

Reply:
We can remove it for now.

Reply:
it's because attributes need a UID, but we can later make attribute code as ID type

Reply:
```suggestion
```

Reply:
Removing attribute_uid after suggestions for @cpartica and @melnikovi. Thanks @mslabko for pointing it out.